nav2_localization/src/nav2_localization.cpp

```diff
@@ -0,0 +1,242 @@
+#include <chrono>
+
+#include "nav2_localization/nav2_localization.hpp"
+#include "nav2_util/string_utils.hpp"
+#include "tf2_ros/create_timer_ros.h"
+
+using namespace std::chrono_literals;
+
+namespace nav2_localization
+{
+
+LocalizationServer::LocalizationServer()
+: LifecycleNode("localization_server", "", true),
+  sample_motion_model_loader_("nav2_localization", "nav2_localization_base::SampleMotionModel"),
+  default_sample_motion_model_id_("DummyMotionSampler"),
+  matcher2d_loader_("nav2_localization", "nav2_localization_base::Matcher2D"),
+  default_matcher2d_id_("DummyMatcher2D"),
+  solver_loader_("nav2_localization", "nav2_localization_base::Solver"),
+  default_solver_id_("DummySolver")
+{
+    RCLCPP_INFO(get_logger(), "Creating localization server");
+
+    declare_parameter("sample_motion_model_id", default_sample_motion_model_id_);
+    declare_parameter("matcher2d_id", default_matcher2d_id_);
+    declare_parameter("solver_id", default_solver_id_);
+    declare_parameter("first_map_only", true);
+    declare_parameter("laser_scan_topic_", "scan");
+    declare_parameter("odom_frame_id", "odom");
+    declare_parameter("alpha1", 0.2);
```

Comment:
Make sure that plugin specific parameters don't get found in the main server, they should be gotten in their own specific plugins so that they're portable and we don't need to expose every parameter for every possible plugin in the application. I think the `alpha`s here are the only example of that, the rest are globally needed items.